### PR TITLE
Narrow filters for emitting new docs

### LINKED ISSUE #1932: Narrow filters for emitting new docs
## Problem

Currently the documentation is rebuilt if any file under the `documentation/` or `frontend/` directories has changed. While this is acceptable for the former, the latter changes very frequently without the documentation changing.

## Description

By updating the [filter values for `documentation`](https://github.com/WordPress/openverse/blob/403cdf65420040531b5af1c5852957c8e13ca103/.github/filters.yml#L22) to include

- `frontend/**.mdx`
- `frontend/.storybook`
- `frontend/nuxt.config.ts`
- `frontend/tailwind.config.js`

we can drop the checks for the frontend changes and only build docs if either the Sphinx site, or the TCV, or the Storybook has changed. This is somewhat related to #475.

## Additional context

The goal of this is speed up CI for the frontend, by preventing the need to build documentation for each PR.

### PR DESCRIPTION
**Pull request by veronewra**
## Fixes

Fixes #1932 by @dhruvkb

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by veronewra** [comment]
sorry this is my first time contributing to openverse- it looks like it failed some checks. I would love some feedback and guidance

**Comment by AetherUnbound** [comment]
Thanks and congrats on your first contribution @veronewra! Those are checks that maintainers need to address regarding proper labeling of the PR - I've gone ahead and fixed them 😄

**Comment by AetherUnbound** [comment]
@dhruvkb is this the set of changes you were imagining when describing #1932?

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Before a more thorough review, the `frontend` filters should not change, only the fillets for `documentation`.

**Review by obulat — DISMISSED** [review]
Welcome to Openverse, @veronewra! 😺 

Could you re-add the frontend parts (lines 13-21)?

Also, I think we should add these after line 18, because it can also change the documentation:
`frontend/src/styles/tailwind.css`

File: .github/filters.yml [hunk 1]
```diff
@@ -10,17 +10,12 @@ ingestion_server:
   - ingestion_server/**
   # Change to the CI + CD workflow should trigger complete workflow.
   - .github/workflows/ci_cd.yml
```
Comment (obulat) [inline .github/filters.yml:12]:
```suggestion
  - .github/workflows/ci_cd.yml
  frontend:
  - frontend/**
  - package.json
  - pnpm-lock.yaml
  - pnpm-workspace.yaml
  - .npmrc
  - .pnpmfile.cjs
  # Change to the CI + CD workflow should trigger complete workflow.
  - .github/workflows/ci_cd.yml
```

File: .github/filters.yml [hunk 2]
```diff
@@ -10,17 +10,12 @@ ingestion_server:
...
-  - .npmrc
-  - .pnpmfile.cjs
-  # Change to the CI + CD workflow should trigger complete workflow.
-  - .github/workflows/ci_cd.yml
 documentation:
   - documentation/**
+  - frontend/**.mdx
+  - frontend/.storybook
+  - frontend/nuxt.config.ts
+  - frontend/tailwind.config.js
```
Comment (obulat) [inline .github/filters.yml:18]:
```suggestion
  - frontend/tailwind.config.js
  - frontend/src/styles/tailwind.css
```

**Comment by dhruvkb** [comment]
#1933 might be a blocker/conflict-creator for this PR. Just noting.

File: .github/filters.yml [hunk 3]
```diff
@@ -10,17 +10,12 @@ ingestion_server:
...
-  - pnpm-lock.yaml
-  - pnpm-workspace.yaml
-  - .npmrc
-  - .pnpmfile.cjs
-  # Change to the CI + CD workflow should trigger complete workflow.
-  - .github/workflows/ci_cd.yml
 documentation:
   - documentation/**
+  - frontend/**.mdx
+  - frontend/.storybook
```
Comment (dhruvkb) [inline .github/filters.yml:16]: I don't think `filters.yml` works with directories.
```suggestion
  - frontend/.storybook/**
```

File: .github/filters.yml [hunk 4]
```diff
@@ -10,17 +10,12 @@ ingestion_server:
...
+  - frontend/nuxt.config.ts
+  - frontend/tailwind.config.js
   # Change to the CI + CD workflow should trigger complete workflow.
   - .github/workflows/ci_cd.yml
```
Comment (dhruvkb) [inline .github/filters.yml:20]: Drop these lines w.r.t. #1933

```suggestion
  # Change to the CI + CD workflow should trigger complete workflow.
  - .github/workflows/ci_cd.yml
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
You will also need to update the CI + CD workflow.
image

**Comment by veronewra** [comment]
Right! Should I remove all instances of "needs.get-changes.outputs.frontend == 'true' " in the ci_cd.yml file?

**Comment by dhruvkb** [comment]
Not all @veronewra, these 2 specifically:

https://github.com/WordPress/openverse/blob/e78b0bc8c1a0b4438912b7843973bbfeb5c877e6/.github/workflows/ci_cd.yml#L702

https://github.com/WordPress/openverse/blob/e78b0bc8c1a0b4438912b7843973bbfeb5c877e6/.github/workflows/ci_cd.yml#L921

**Review by dhruvkb — APPROVED** [review]
Awesome, LGTM! Thanks for this contribution @veronewra.

**Review by zackkrida — APPROVED** [review]

**Comment by veronewra** [comment]
@dhruvkb @AetherUnbound I want to join the slack so I can ask questions /attend office hours, but when I try to make a wordpress account it does not send me a confirmation email (I checked spam and all). Is there a way I can directly join the slack? Thanks:)

**Comment by AetherUnbound** [comment]
@veronewra we'd love to have you! It's a little complex to get added, check https://make.wordpress.org/chat/ for instructions (you'll need to be logged into a WordPress.org account first). Here's the instructions I see when I'm logged in:

**Pull request merged**